### PR TITLE
[TFLite][CI] Update TensorFlow dependency to 2.9.1

### PR DESCRIPTION
**Pull request by leandron**
This updates the TF version to be used in TVM CI to 2.9.1, which brings improvements so that more platforms are supported by official packages. This PR updates the Docker images scripting to install TF and TFLite and some CMake changes required when updating from TF 2.6.x to 2.9.x.

Adding it here as a draft until we get #12130 merged.

cc @Mousius @NicolaLancellotti @areusch @driazati @gigiblender

### CODE REVIEW

**Comment by areusch** [comment]
@leandron when investigating the Python deps I also came across this bug. We currently build a tflite library from the tensorflow source, and in TF 2.7 the build system was switched to CMake I believe. I haven't yet had time to solve this, but I'll let you know if I do.

**Comment by leandron** [comment]
> @leandron when investigating the Python deps I also came across this bug. We currently build a tflite library from the tensorflow source, and in TF 2.7 the build system was switched to CMake I believe. I haven't yet had time to solve this, but I'll let you know if I do.

This is solved in this PR. The only issue is that the CI is testing with the old images. I'll generate a new set from `ci-docker-staging` and re-run the tests. it should be fine.

**Comment by areusch** [comment]
hm, was that https://ci.tlcpack.ai/blue/organizations/jenkins/docker-images-ci%2Fdocker-image-run-tests/detail/docker-image-run-tests/171/pipeline?

**Comment by leandron** [comment]
Yes. However it still requires the Docker images to have the updated TF for the whole CI.

I generated a new set, and will make a new CI staging run with the new images. Then it should be fine.

**Comment by areusch** [comment]
ah ok cool--lmk when i can take a look at them

**Comment by leandron** [comment]
@areusch just to update here as we're discussing, once this is merged, the PR that updates the Docker images, will need to update `task_config_build_cpu.sh` from:
```
echo set\(USE_TFLITE ON\) >> config.cmake
```

to:

```
echo set\(USE_TFLITE /opt/tflite\) >> config.cmake
```

This way we keep the default source based installation in the new format introduced here.

**Comment by areusch** [comment]
@mbrookhart could you have a look?

File: docker/Dockerfile.ci_cpu
Comment (mbrookhart) [inline docker/Dockerfile.ci_cpu:143]: Little concered about a comment that says "do not push to Github" in a github PR...

File: docker/install/ubuntu_install_cmake_source.sh
Comment (mbrookhart) [inline docker/install/ubuntu_install_cmake_source.sh:23]: Do we need to update the minimum cmake version in the cmake file?

**Review by mbrookhart — COMMENTED** [review]
A couple of minor comments, but overall I think I'm okay with this.

File: docker/install/ubuntu_install_cmake_source.sh
Comment (leandron) [reply docker/install/ubuntu_install_cmake_source.sh:23]: No because the update is only required by tflite build, not TVM

File: docker/install/ubuntu_install_tflite.sh
Comment (driazati) [inline docker/install/ubuntu_install_tflite.sh:49]:
```suggestion
cmake --build . -- -j"$(nproc)"
```

File: docker/install/ubuntu_install_cmake_source.sh
Comment (driazati) [reply docker/install/ubuntu_install_cmake_source.sh:23]: does tflite require 3.22? #12682 is also trying to update the CMake version to 3.18 and we should pick the lower of the two for what runs in CI (if possible)

Comment (leandron) [reply docker/install/ubuntu_install_cmake_source.sh:23]: 3.18 is also accepted by tflite. Will do the change here and notify.

File: docker/Dockerfile.ci_cpu
Comment (leandron) [reply docker/Dockerfile.ci_cpu]: Oops! Removed now. For context: this we needed before I realised that `ethos-u-vela` needed to be upgraded, which had a hard requirement on an old numpy. With that, this workaround was not needed anymore.

File: docker/install/ubuntu_install_cmake_source.sh
Comment (driazati) [inline docker/install/ubuntu_install_cmake_source.sh:23]: looks like this also needs to change to get a valid URL below
```suggestion
v=3.18
```

**Review by driazati — APPROVED** [review]
lgtm once comments are addressed

**Comment by leandron** [comment]
> lgtm once comments are addressed

Thanks @driazati

**Comment by leandron** [comment]
@tvm-bot rerun

**Comment by leandron** [comment]
@tvm-bot rerun

**Comment by leandron** [comment]
> Looks good to me, just wanted to mention that this might break microTVM testing on hardware since RVM uses a different tensorflow version. If it does we need a follow up PR.
> 
> https://github.com/apache/tvm/blob/2622ac9e638b259cae017813ad93937c0ff8a2f9/apps/microtvm/pyproject.toml#L107

Did this change recently? I’ve run many test rounds and haven’t seen any breakage in microtvm

**Comment by areusch** [comment]
thanks @leandron !!

**Comment by mehrdadh** [comment]
@leandron I don't think it changed recently, it should be fine

**Pull request merged**
